src/Microsoft.ML.OnnxTransformer/OnnxTransform.cs

```diff
@@ -537,15 +550,16 @@ private void UpdateCacheIfNeeded(long position, INamedOnnxValueGetter[] srcNamed
                         outputCache.Outputs[outputNameOnnxValue.Name] = outputNameOnnxValue;
                     }
                     outputCache.Position = position;
+                    outputCache.NamedOnnxValues = outputNamedOnnxValues;
                 }
             }
 
             private Delegate MakeTensorGetter<T>(DataViewRow input, int iinfo, INamedOnnxValueGetter[] srcNamedValueGetters, string[] activeOutputColNames)
             {
                 Host.AssertValue(input);
-                var outputCacher = new OnnxRuntimeOutputCacher();
                 ValueGetter<VBuffer<T>> valueGetter = (ref VBuffer<T> dst) =>
                 {
+                    using var outputCacher = new OnnxRuntimeOutputCacher();
```

Comment:
Are we even "caching" anything anymore if you move this inside of the valueGetter delegate? A new one is going to be created for each call of this delegate.

You may want to make sure this isn't slowing things down drastically.

Reply:
Actually, Michael and I are just discussing this. I will post back an update soon.